### ISSUE TITLE
Extending the XMPP page

### ISSUE THREAD

**Issue by drank40** [LukeSmithxyz/landchad#88]
Hello,

I have followed the guide on how to setup a prosody server with great success. Everything works as its supposed to, but i feel like it would need file sharing that isn't p2p and that also works in MUCs to be used practically. So far I haven't found plenty of resources on how to implement this through extensions, yet I know it is possible. 
Maybe this could be included in a separate page.

**Comment by artur-borys** [comment]
According to [``mod_http_file_share``](https://prosody.im/doc/modules/mod_http_file_share), http file sharing will be available in the next major release of prosody, however I don't know if it will be e2e encrypted. There is something going on with OMEMO file sharing, (https://xmpp.org/extensions/inbox/omemo-media-sharing.html), however it isn't officially being worked on yet, so there's probably not much we can do. I've tried to look for something, but haven't found anything useful

**Comment by drank40** [comment]
Thanks, I wonder if ejabberd has a better implementation for file sharing. I see that ejabberd is listed on the "On the look out for..." section, I'll try and have a look at it.

**Comment by drank40** [comment]
Update : the feauture works well on trunk builds of prosody.

**Comment by TheYellowArchitect** [comment]
@drank40 did you install prosody and it worked by default? I installed prosody like a month ago, and tried to install file transfer, I literally failed and I don't know what is even at fault.
tl;dr: I upload a file and it says file transfer failed

imo I see file transferring as a necessity in modern communications (its not just normies, imagine not being able to transfer a meme or a .zip), and the tutorial needs file upload if it doesn't come by default imo

**Comment by Abdulee** [comment]
Yes, I think that prosody page should be extended or maybe he shows how to install new modules.. And maybe setting up a turn, stun server for calling .. I managed to get it done today so I will probably write and submit a pr as a new page..

**Comment by drank40** [comment]
@TheYellowArchitect i have to say it was very intracate. First of all you have to install the mercurial version (the trunk version) then you have to get extra community plugins and add them to your prosody.conf.lua. I think it would be better if you posted your steps, what works for you and what doesnt in order for me to help you and possibly draft a guide to reproduce it. The main challenges are getting your certs setup correctly for all your subdomains, and actually enabling the extension

**Comment by drank40** [comment]
Here is my prosody.cfg.lua, stripped of any domain and user info. Replacing example.com with your domain and your admins should make it usable
[prosody.cfg.lua.txt](https://github.com/LukeSmithxyz/landchad/files/7271619/prosody.cfg.lua.txt)

**Comment by drank40** [comment]
A good resource to test your server is https://compliance.conversations.im/
